### PR TITLE
feat: new offer submission price range notice

### PR DESCRIPTION
**Pull request by tam-kis**
[NX-3013]
Includes warning message when the custom price is too low, autoselect when clicking the notice, as well as fixes in analytics and copy.

Screenshot 2021-12-23 at 18 32 41

cc @artsy/negotiate-devs 

[NX-3013]: https://artsyproduct.atlassian.net/browse/NX-3013?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jo-rs** [comment]
Current status:
autoselect

Missing:
- Tests
- Potentially some refactoring

File: src/v2/Apps/Order/Components/OfferNote.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const OfferNote: React.FC<{
   onChange(change: TextAreaChange): void
   artworkId: string
   counteroffer?: boolean
-}> = ({ onChange, artworkId, counteroffer }) => {
```
Comment (jo-rs) [inline src/v2/Apps/Order/Components/OfferNote.tsx:9]: @tam-kis why did you remove this? 🤔

File: src/v2/Apps/Order/Components/PriceOptions.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { BorderedRadio, Flex, RadioGroup, Text } from "@artsy/palette"
 import { useEffect, useState } from "react"
 import { OfferInput } from "v2/Apps/Order/Components/OfferInput"
+import { MinPriceWarning } from "./MinPriceWarning"
```
Comment (jo-rs) [inline src/v2/Apps/Order/Components/PriceOptions.tsx:4]: Moved the min price notice to its own component since this one was getting chonky

File: src/v2/Apps/Order/Components/PriceOptions.tsx [hunk 3]
```diff
@@ -47,48 +59,37 @@ export const PriceOptions: React.FC<PriceOptionsProps> = ({
...
       description: rangePrice.description,
     }))
   }
 
   const getPercentageOptions = () => {
-    return [0.2, 0.15, 0.1].map(pricePercentage => {
+    return [0.2, 0.15, 0.1].map((pricePercentage, idx) => {
       if (listPrice?.major) {
         return {
+          key: `price-option-${idx}`,
```
Comment (jo-rs) [inline src/v2/Apps/Order/Components/PriceOptions.tsx:92]: Used for the auto select

File: src/v2/Apps/Order/Components/PriceOptions.tsx [hunk 4]
```diff
@@ -100,26 +101,42 @@ export const PriceOptions: React.FC<PriceOptionsProps> = ({
...
+        flow: AnalyticsSchema.Flow.MakeOffer,
+        order_id: order.internalID,
+      })
+    }
+  }
 
-  // TODO: add call bellow when the feature is implemented
-  // trackClick("We recommend changing your offer", priceOptions[0]?.value!)
+  const selectMinPrice = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.stopPropagation()
```
Comment (jo-rs) [inline src/v2/Apps/Order/Components/PriceOptions.tsx:118]: Stops the `onSelect` from getting triggered

File: src/v2/Apps/Order/Components/PriceOptions.tsx [hunk 5]
```diff
@@ -100,26 +101,42 @@ export const PriceOptions: React.FC<PriceOptionsProps> = ({
...
+    trackClick("We recommend changing your offer", priceOptions[0]?.value!)
+    setSelectedRadio("price-option-0")
+    setToggle(false)
+    setCustomValue(undefined)
+    onChange(minPrice)
+  }
 
   return (
-    <RadioGroup>
+    <RadioGroup onSelect={setSelectedRadio} defaultValue={selectedRadio}>
```
Comment (jo-rs) [inline src/v2/Apps/Order/Components/PriceOptions.tsx:127]: Both needed for the auto select to be possible when clicking the notice.

File: src/v2/Apps/Order/Components/OfferNote.tsx (hunk 1 above)
Comment (jo-rs) [reply src/v2/Apps/Order/Components/OfferNote.tsx:9]: Oh, I see it was not being used. Weird.

File: src/v2/Apps/Order/Components/MinPriceWarning.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  }, [])
+
+  return (
+    <Message variant="default" p={2} mt={2}>
+      Galleries usually accept offers within
+      {isPriceRange ? " the displayed price range" : " 20% of the listed price"}
+      ; any lower is likely to be rejected.
+      <br />
+      <Clickable textDecoration="underline" cursor="pointer" onClick={onClick}>
+        {`We recommend changing your offer to ${minPrice}.`}
```
Comment (lean8086) [inline src/v2/Apps/Order/Components/MinPriceWarning.tsx:34]: Out of curiosity, not necessarily asking for a change: Is there a reason to use a template literal for the copy instead of smt like:
```
<Clickable ...>We recommend changing your offer to {minPrice}.</Clickable>
```

File: src/v2/Apps/Order/Components/PriceOptions.tsx [hunk 7]
```diff
@@ -143,19 +149,31 @@ export const PriceOptions: React.FC<PriceOptionsProps> = ({
               trackClick("Different amount", 0)
             }}
             key="price-option-custom"
+            name="radio"
```
Comment (lean8086) [inline src/v2/Apps/Order/Components/PriceOptions.tsx:152]: Is a `name` needed here? I couldn't find where it's been used. If it's needed, would it make sense to use a more descriptive name?

File: src/v2/Apps/Order/Components/MinPriceWarning.tsx (hunk 6 above)
Comment (jo-rs) [reply src/v2/Apps/Order/Components/MinPriceWarning.tsx:34]: Heh no, none. Oops. Correcting.

File: src/v2/Apps/Order/Components/PriceOptions.tsx (hunk 7 above)
Comment (jo-rs) [reply src/v2/Apps/Order/Components/PriceOptions.tsx:152]: I forgot if I added these for it to work well with RTL, but I'll take a look.

Comment (lean8086) [reply src/v2/Apps/Order/Components/PriceOptions.tsx:152]: I saw RTL is [getting them by role](https://github.com/artsy/force/pull/9118/files#diff-e3c631e6b08c80b65bdfee8975885955c11cf9850f8fb85bfc7e38657b1e10a9R154).

Comment (jo-rs) [reply src/v2/Apps/Order/Components/PriceOptions.tsx:152]: Yeah, I must've left it there by mistake. Removing.

**Review by lean8086 — APPROVED** [review]
LGTM 🚀 Thanks for the comments on the code, it made reviewing it easier!

File: src/v2/Apps/Order/Routes/Offer/index.tsx [hunk 8]
```diff
@@ -272,7 +272,7 @@ export class OfferRoute extends Component<OfferProps, OfferState> {
               )}
               {newOfferSubmissionEnabled && !isInquiryCheckout && !isEdition && (
                 <>
-                  <Text variant="lg" color="black80" marginTop={4}>
+                  <Text variant="lg" color="black80" marginTop={2}>
```
Comment (tam-kis) [inline src/v2/Apps/Order/Routes/Offer/index.tsx:275]: We should change the `marginTop` to `mt` if that's the correct one to use, as in line `154`
